app/frontend/src/features/communities/leaveReference/ReferenceDataContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+export type ReferenceContextFormData = {
+  text: string;
+  wasAppropriate: string;
+  rating: number;
+};
+
+interface ReferenceDataProviderProps {
+  children: ReactNode;
+}
+
+type ReferenceContextType = {
+  data: ReferenceContextFormData;
+  setValues: (data: ReferenceContextFormData) => void;
+};
+
+const ReferenceDataContext = createContext<ReferenceContextType | undefined>(
+  undefined
+);
+
+export const ReferenceDataProvider = ({
+  children,
+}: ReferenceDataProviderProps) => {
+  const [data, setData] = useState({
+    text: "",
+    wasAppropriate: "",
+    rating: 0,
+  });
+
+  const setValues = (values: any) => {
+    setData((prevData) => ({
+      ...prevData,
+      ...values,
+    }));
+  };
+
+  return (
+    <ReferenceDataContext.Provider value={{ data, setValues }}>
```

Comment:
I'm not really sure this really needs to be in an context - you can easily pass this down as props from the reference page component. It's better to not reach for it straight away unless you have an unavoidable and deep prop drilling problem.

Also passing multiple values with the state and setter together tend to cause more unnecessary renders and it's this pattern that usually kill React app perf and people complain about context being slow, so I'd rethink about the need to use it.